docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -242,9 +242,369 @@ This guide will help you create a new Expectation Suite by profiling your data w
+                                 #      domains are checked against these types for inclusion in the rule domain.
+                                 #      All domains corresponding to any of the semantic_types are included.
+                                 # semantic_type key is determined by the domain builder not global e.g. SimpleSuffixColumnDomainBuilder where suffix is the keyword.
+                  - datetime
+                user_input_list_of_domain_names: # TBD - this is maybe batch_id,domain_id together - probably Typed Object DomainId - can add human readable IDs. Probably too much complexity for first version.
+              parameter_builders:
+                - name: my_dateformat  # This is shorthand for `parameter_builder_name`
+                  # JPC: what is happening here -- we're asking, "What date format matches the data in this column?"
+                  class_name: SimpleDateFormatStringParameterBuilder
+                  module_name: # OPTIONAL
+                  domain_builder_kwargs: my_domain_builder_name.domain_kwargs # QUESTION: if there is only one domain_builder, can this just be assumed?
+              expectation_configuration_builders:
+                # TODO: BranchingExpectationConfigurationBuilder - with parameters that take key value for gt lt gte lte
+                # Keep this if-then complexity in the Expectation rather than in the config
+                # Propose delaying until later versions, but this is directional design
+
+                - branch: # branch is optional if/then syntax. You can also just put the expectation
+                    if: $my_dateformat.success_ratio >= 0.8  # if evaluates to true in python, i.e. "" is FALSE, "%Y" is TRUE
+                                                                   # This success_ratio is provided via `details` of a parameter
+                                                                   # It is matched with the correct domain
+                                                                   # QUESTION: I removed the `parameter` in between the
+                                                                   # parameter_name and success_ratio - perhaps we can
+                                                                   # simplify these lookups - ANSWER - Cool, not sure exaclty how to implement but let's try it.
+                    then:
+                      # ExpectationConfigurationBuilder configuration goes here
+                      - name: my_expectation_configuration_builder_1
+                        class_name: DefaultExpectationConfigurationBuilder # Optional, will use Default if not supplied, otherwise will use supplied
+                        module_name: tbd # Optional, for all other
+                        expectation_type: expect_column_values_to_match_strftime_format # QUESTION: Should we change this from expectation to expectation_type?
```

Comment:
My opinion is that expectation sounds more intuitive because expectation may insinuate table expectation, column expectation, etc.

Reply:
I think so too, but let's keep it `expectation_type` to be consistent with other areas of the codebase and propose to change it wholesale rather than be inconsistent.

Reply:
Actually I kind of like `expectation_name` as well but could be swayed.